torch/csrc/jit/codegen/cuda/lower_validation.cpp

```diff
@@ -96,10 +77,10 @@ void IRReplaceSizes() {
   // TensorView wouldn't change, so users pointers will remain valid. The other
   // option which seems less elegant but would also work is build up the domain
   // on the new tensor, and then simply replace it into the original one.
-  for (TensorView* tv : orig_inp_out) {
+  for (auto tv : inputs_and_outputs) {
     // Replace the domain with one based on Ti.size[j]
     std::vector<IterDomain*> new_domain_iters;
-    const std::vector<IterDomain*>& root_td = tv->getRootDomain();
```

Comment:
why?

Reply:
just a minor simplification (I think I rewrote more of this code, then reverted most of it)

If you don't like it I'll revert this change.

Reply:
On second review the auto on the for loop argument is fine as the name pretty strictly implies it's a TensorView. The auto on the root_td I dislike, as the naming is conflicting with types. getRootDomain explicitly does not return a TensorDomain and it's important to keep in mind that this type is a vector of IterDomains, not a TensorDomain.

Generally I do not prefer auto over stricter typing and myself should use auto more sparingly. Many people still don't use modern IDEs and figuring out types can be painful.

Reply:
reverted

Reply:
> The auto on the root_td I dislike, as the naming is conflicting with types. getRootDomain explicitly does not return a TensorDomain and it's important to keep in mind that this type is a vector of IterDomains, not a TensorDomain.

This is a good point. Would it make sense to rename getRootDomain() (and TendorDomain::rootDomain())? Or make them return TensorDomain?